garbagecoll/collector.go

```diff
@@ -0,0 +1,87 @@
+package garbagecoll
+
+import (
+	"context"
+	"github.com/antonbaumann/spotify-jukebox/db"
+	"github.com/sirupsen/logrus"
+	"time"
+)
+
+const (
+	SessionExpiration = time.Hour * 48
+	CleaningInterval  = time.Hour
+)
+
+// responsible for deleting inactive sessions
+type garbageCollector struct {
+	ticker            *time.Ticker
+	userCollection    db.UserCollection
+	sessionCollection db.SessionCollection
+	quit              chan bool
+}
+
+func New(users db.UserCollection, sessions db.SessionCollection) *garbageCollector {
+	return &garbageCollector{
+		ticker:            time.NewTicker(CleaningInterval),
+		userCollection:    users,
+		sessionCollection: sessions,
+	}
+}
+
+func (gc garbageCollector) Run() {
+	// quit chanel should only exist when gc is running
+	quit := make(chan bool)
+	gc.quit = quit
+
+	go func() {
+		for {
+			select {
+			case <-gc.ticker.C:
+				gc.clean()
+			case <-quit:
+				gc.ticker.Stop()
+				gc.quit = nil
+				return
+			}
+		}
+	}()
+}
+
+func (gc garbageCollector) Quit() {
+	if gc.quit == nil {
+		logrus.Warn("garbage collector not running")
+		return
+	}
+	gc.quit <- true
+}
+
+func (gc garbageCollector) clean() {
+	msg := "[garbagecoll] clean"
+	ctx := context.Background()
+
+	sessionIDs, err := gc.sessionCollection.ListSessionIDs(ctx)
```

Comment:
I think it is possible to write a Mongo query that deletes all sessions that should be deleted in one single operation